internal/view/ns.go

```diff
@@ -99,3 +102,74 @@ func (n *Namespace) decorate(data *render.TableData) {
 		}
 	}
 }
+
+func (n *Namespace) deleteNamespaceKeyBindings(evt *tcell.EventKey) *tcell.EventKey {
+	app := n.App()
+	cfg := app.Config
+	cl := cfg.K9s.ActiveCluster()
+
+	if cl == nil {
+		return evt
+	}
+
+	styles := app.Styles.Dialog()
+	bindingsToDelete := map[string]struct{}{}
+
+	f := tview.NewForm().
+		SetItemPadding(0).
+		SetButtonsAlign(tview.AlignCenter).
+		SetButtonBackgroundColor(styles.ButtonBgColor.Color()).
+		SetButtonTextColor(styles.ButtonFgColor.Color()).
+		SetLabelColor(styles.LabelFgColor.Color()).
+		SetFieldTextColor(styles.FieldFgColor.Color()).
+		SetFieldBackgroundColor(styles.BgColor.Color())
+
+	for _, namespaceName := range cl.Namespace.Favorites {
+		// The browser already reserves Key0 to the `all` namespace
+		// shortcut, so we only allow deletion of other favorites namespaces
+		if client.IsAllNamespace(namespaceName) {
+			continue
+		}
+
+		f.AddCheckbox(namespaceName, false, func(label string, _ bool) {
+			bindingsToDelete[label] = struct{}{}
+		})
+	}
+
+	f.AddButton("Cancel", func() {
+		app.Content.RemovePage(deleteNumericBindingsKey)
+	}).AddButton("Ok", func() {
+		app.Content.RemovePage(deleteNumericBindingsKey)
+
+		if len(bindingsToDelete) == 0 {
+			return
+		}
+
+		// Rebuild list of favorite namespaces based on what was marked for deletion
+		newFavorites := make([]string, 0)
+
+		for _, favNsName := range cl.Namespace.Favorites {
+			if _, ok := bindingsToDelete[favNsName]; ok {
+				continue
+			}
+
+			newFavorites = append(newFavorites, favNsName)
+		}
+
+		cl.Namespace.Favorites = newFavorites
+	})
+
+	for i := 0; i < f.GetButtonCount(); i++ {
+		if b := f.GetButton(i); b != nil {
+			b.SetBackgroundColorActivated(styles.ButtonFocusBgColor.Color())
+			b.SetLabelColorActivated(styles.ButtonFocusFgColor.Color())
+		}
+	}
+
+	modal := tview.NewModalForm("Delete Numeric Bindings", f)
+	app.Content.Pages.AddPage(deleteNumericBindingsKey, modal, false, true)
```

Comment:
Need to implement SetDoneFunc so users can escape out ;(

Reply:
this is no longer available as I no lunger use the delete ns function